### PR TITLE
chore: bump project-factory to ~> 11

### LINKED ISSUE #129: Dependency failures on M1 Macs
```
╷
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/random v2.3.1 does not have a package available for your current platform, darwin_arm64.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have different platforms supported.
╵

╷
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/null v2.1.2 does not have a package available for your current platform, darwin_arm64.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have different platforms supported.
╵
```

These come in from `project-factory`, which should be updated to [v11.0.0+](https://github.com/terraform-google-modules/terraform-google-project-factory/pull/583) to get the above to versions that support ARM.

### PR DESCRIPTION
**Pull request by cagataygurturk**
Fixes #129

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by cagataygurturk** [comment]
I am not sure why integration tests did not pass. Locally they did.

File: main.tf [hunk 1]
```diff
@@ -56,7 +56,7 @@ resource "google_folder_iam_member" "tmp_project_creator" {
 
 module "seed_project" {
   source                      = "terraform-google-modules/project-factory/google"
-  version                     = "~> 10.1.1"
+  version                     = "~> 11"
```
Comment (morgante) [inline main.tf:59]: We pin to a major version.

```suggestion
  version                     = "~> 11.2"
```

File: modules/cloudbuild/main.tf [hunk 2]
```diff
@@ -39,7 +39,7 @@ data "google_organization" "org" {
 
 module "cloudbuild_project" {
   source                      = "terraform-google-modules/project-factory/google"
-  version                     = "~> 10.1.1"
+  version                     = "~> 11"
```
Comment (morgante) [inline modules/cloudbuild/main.tf:42]:
```suggestion
  version                     = "~> 11.2"
```

**Review by morgante — CHANGES_REQUESTED** [review]

**Review by morgante — APPROVED** [review]

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**